builder/builder.go

```diff
@@ -524,26 +532,50 @@ func (b *Builder) BuildChroots(template *x509.Certificate, privkey *rsa.PrivateK
 
 // BuildUpdate will produce an update consumable by the swupd client
 func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string, skipSigning bool, publish bool, keepChroots bool) error {
+	var err error
+
+	if minVersion < 0 || minVersion > math.MaxUint32 {
```

Comment:
If we had a type for "version", we could specify that minVersion was of this type and remove this check.

Reply:
Agreed. Either that or simply using Uint32.

I had to pick my battles since this would cause a lot of changes outside the scope of what I was doing.